File: PhysicsTools/LiteAnalysis/bin/testPAT.cc
//-----------------------------------------------------------------------------
// File: testPAT.cc
// Created: 24-Jun-2007 Harrison B. Prosper
// Updated: 01-Nov-2007 HBP
//-----------------------------------------------------------------------------
#include "PhysicsTools/LiteAnalysis/interface/EventStream.h"
#include "PhysicsTools/LiteAnalysis/interface/kit.h"
#include <stdio.h>
//-----------------------------------------------------------------------------
#include "DataFormats/EgammaReco/interface/SuperCluster.h"
#include "DataFormats/TrackReco/interface/Track.h"
#include "DataFormats/EgammaCandidates/interface/GsfElectron.h"
#include "DataFormats/JetReco/interface/CaloJet.h"
#include "DataFormats/Math/interface/Vector3D.h"
#include "DataFormats/PatCandidates/interface/Jet.h"
#include "FWCore/FWLite/interface/AutoLibraryLoader.h"
#include "HepMC/GenParticle.h"
//-----------------------------------------------------------------------------
#include <TROOT.h>
#include <TSystem.h>
#include <TLorentzVector.h>
#include <TApplication.h>
//-----------------------------------------------------------------------------
#include <string>
#include <vector>
#include <fstream>
using namespace std;
//-----------------------------------------------------------------------------

//-----------------------------------------------------------------------------
void fatal(string message)
{
  cout << "** error ** " << message << endl;
  exit(0);
}
//-----------------------------------------------------------------------------
int 
main(int argc, char** argv)
{
  cout << "\n\ttestPAT\n" << endl;

  AutoLibraryLoader::enable();

  string filename("ZEE_PAT.root");

  EventStream stream(filename);
  if ( !stream.good() ) fatal("unable to open file");
    
  int nevent = stream.size();
  cout << "\tnumber of events: " << nevent << endl;

  vector<pat::Jet> jets;    
  stream.select(jets);
  if ( !stream.good() ) fatal("problem getting branch");
    
  char record[80];
  sprintf(record, "%4s\t%4s", "Row", "Njet");
  cout << record << endl;

  for(int row=0; row <  10; row++)
    {
      stream.read(row);
      sprintf(record, "%4d\t%4d", row, jets.size());
      cout << record << endl;
    }
}

File: PhysicsTools/LiteAnalysis/test/eventstream/testPAT.py
#!/usr/bin/env python
#-----------------------------------------------------------------------------
# Filename: testPAT.py
# Description:
# Created:  15-Sep-2008 HBP
#-----------------------------------------------------------------------------
import os
import sys
from sys import exit
from ROOT import *
from math import *
from string import *
from PhysicsTools.LiteAnalysis import *
#-----------------------------------------------------------------------------
# Constants
#-----------------------------------------------------------------------------
def main():
	print "\n\ttestAOD\n"

	filename = "zee_pat.root"

	print "\topening file %s\n" % filename
	stream = EventStream(filename)
	if not stream.good(): fatal("unable to open file")

	nevent = stream.size()
	print "\tnumber of events: %d\n" % nevent

	electrons = vector("pat::Electron")()
	stream.select(electrons)

	jets = vector("pat::Jet")()
	stream.select(jets)

	genparticles = vector("reco::GenParticle")()
	stream.select(genparticles)

	print "%4s\t%10s%10s%10s" % ("Row", "Npart", "Nele", "Njet")
	
	for (row, stat) in enumerate(stream):

		print "%4d\t%10d%10d%10d" % (row,
									 genparticles.size(),
									 electrons.size(),
									 jets.size())
		if row > 99: break
#---------------------------------------------------------------------------
main()




